Expand vc api details (#37)
credential-handler/chapi.io: developers/wallets/native/index.md

Now that your native wallet has been registered with CHAPI, it can receive requests to store credentials at the URL stated in the `credential_handler.url` of the `manifest.json` above.

CHAPI will use the endpoint declared (`/switchboard` in the example above) when it provides links to available wallets in the user's browser:

> ![Choose a wallet modal presenting all preregisterd wallet systems which can be clicked on to proceed to storing the credentials there.](/images/VeresCHAPIaccept.png)

Reviewer: ```suggestion
![Choose a wallet modal presenting all preregistered wallet systems which can be clicked on to proceed to store the credentials there.](/images/VeresCHAPIaccept.png)
```

Reviewer: ah realize this is just the alt text but still worth fixing up

Author: Alt text is still valuable text. ❤️

> The user will click the provided link to selection your Wallet from the list.

Reviewer: ```suggestion
The user will click the provided link to select your wallet from the list.
```

There are currently two possible protocols that Issuers may use with CHAPI in order to issue Verifiable Credentials to be stored in the user's digital wallet: `vcapi` and `OID4VCI`.

> Depending on what you declared in your `acceptedProtocols` you may receive either or both as properties in the `protocols` object carried in the `request` query parameter. Your Wallet's code (i.e. in `/switchboard`) should decide which protocol to use (or it may delegate this choice to advanced users) when multiple protocols are available.

Reviewer: ```suggestion
Depending on what you declared in your `acceptedProtocols` you may receive either or both as properties in the `protocols` object carried in the `request` query parameter. Your wallet's code (i.e. in `/switchboard`) should decide which protocol to use (or it may delegate this choice to advanced users) when multiple protocols are available.
```

### VC-API

> The `vcapi` property will contain a URL that tells the wallet where it can the credential.

Reviewer 2: ```suggestion
The `vcapi` property will contain a URL the wallet can use to get the credential(s).
```

> To retrieve the credential via VC-API, the wallet first sends a `POST` request

Reviewer 2: ```suggestion
To retrieve one or more credentials via VC-API, the wallet first sends a `POST` request
```

to the URL (extracted from `protocols.vcapi` above) with an empty JSON object
(`{}`) in the body of the request to attempt to download the VC to be stored.

```http
POST /exchangers/z1A1GqykGBWKbwhFCDqFjMfnG/exchanges/z19mxa763DAKX7diL51kBFecZ
Host: exchanger.example.com
Content-Type: application/json

{}
```

> The response to that request will contain a Verifiable Credential within a
> `verifiablePresentation` object. Additionally, if there are further steps
> required in the flow such as DID Auth a `verifiablePresentationRequest` object
> will describe the details.

Reviewer 2: ```suggestion
The response to that request will contain one or more verifiable credentials
within a `verifiablePresentation` object if there is anything for the user to
store. If there are further steps required in the flow such as DID Auth a
`verifiablePresentationRequest` object in the response will describe the details.
```

Below is a partial example response:

```json
{
  "verifiablePresentation": {
    "@context": [
      "https://www.w3.org/2018/credentials/v1",
      "https://www.w3.org/2018/credentials/examples/v1",
      "https://w3id.org/security/suites/ed25519-2020/v1"
    ],
    "type": ["VerifiablePresentation"],
    "holder": "did:example:ebfeb1f712ebc6f1c276e12ec21",
    "verifiableCredential": [{

>           "@context": [

Reviewer 3: Fix indent.

        "https://www.w3.org/2018/credentials/v1",
        "https://www.w3.org/2018/credentials/examples/v1",
        "https://w3id.org/security/suites/ed25519-2020/v1"
      ],
      "id": "http://example.edu/credentials/3732",
      "type": [
        "VerifiableCredential",
        "UniversityDegreeCredential"
      ],
      ...
    }]
  }
}
```

More complete examples can be found in the

> [Example Exchanges](https://exchanger.example.com/exchangers/z1A1GqykGBWKbwhFCDqFjMfnG/exchanges/z19mxa763DAKX7diL51kBFecZ)

Reviewer 3: - Looks like the former example URL.  Was that supposed to be preserved somewhere?
- I assume this should now be a spec anchor link?

Author: It'll link to https://w3c-ccg.github.io/vc-api/#exchange-examples in the next update.

section of the [VC-API Specification](https://w3c-ccg.github.io/vc-api/).

### OID4VCI

The `OID4VCI` property will be a URL that includes all of the required information to complete the issuance of the Verifiable Credential using the OID4VCI protocol.

Example OID4VCI protocol URL:
<pre><code style="white-space: normal;overflow: auto;word-break: break-all;">

> `openid-credential-offer://?credential_offer=%7B%22credential_issuer%22%3A%22https%3A%2F%2Fexchanger.example.com%2Fexchangers%2Fz1A1GqykGBWKbwhFCDqFjMfnG%2Fexchanges%2Fz1A36rr6wEL25EEiikKvisVEC%22%2C%22credentials%22%3A%5B%7B%22format%22%3A%22ldp_vc%22%2C%22credential_definition%22%3A%7B%22%40context%22%3A%5B%22https%3A%2F%2Fwww.w3.org%2F2018%2Fcredentials%2Fv1%22%2C%22https%3A%2F%2Fpurl.imsglobal.org%2Fspec%2Fob%2Fv3p0%2Fcontext.json%22%5D%2C%22type%22%3A%5B%22VerifiableCredential%22%2C%22OpenBadgeCredential%22%5D%7D%7D%5D%2C%22grants%22%3A%7B%22urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Apre-authorized_code%22%3A%7B%22pre-authorized_code%22%3A%220065a8a0-069b-46f1-a857-4e1ce5047afd%22%7D%7D%7D`

Reviewer 3: The pr preview looked wrong here. Is this ok live?  Either way, looks like extra backticks around the URL.

Author: My current Markdown highlighter in VIM annoyingly hides certain markup when you're not on the line. Makes it legible...but insane to clean/understand. I need to sort out which thing is doing that...and disable it.

Thanks for the catch!
